data_pipeline/gmail_production_data/get_gmail_data.py

```diff
@@ -0,0 +1,141 @@
+import pandas as pd
+import time
+from functools import wraps
+
+from google.oauth2 import service_account
+from googleapiclient.discovery import build
+
+
+# this connection type will be changed !!
+def connect_to_email(userId):
+    SCOPES = ['https://www.googleapis.com/auth/gmail.readonly']
+    SERVICE_ACCOUNT_FILE = 'DELETE/elife-data-pipeline-test.json'
+    EMAIL_ADDRESS = userId
+
+    credentials = service_account.Credentials.from_service_account_file(
+        SERVICE_ACCOUNT_FILE, scopes=SCOPES)
+    delegated_credentials = credentials.with_subject(EMAIL_ADDRESS)
+
+    service = build("gmail", "v1", credentials=delegated_credentials)
+    return service
+
+
+def retry(ExceptionToCheck, tries=5, delay=4, backoff=3, logger=None):
```

Comment:
I would recommend using https://pypi.org/project/backoff/ we have used it in other Python projects as well.